diffrax/_solver/should.py

```diff
@@ -0,0 +1,239 @@
+        # to 0th power. The descending power order is because of jnp.polyval
+        beta_half = jnp.stack(
+            [-c5 / 3840, c4 / 384, -c3 / 48, c2 / 8, -c / 2, one], axis=-1
+        )
+        beta1 = jnp.stack([-c5 / 120, c4 / 24, -c3 / 6, c2 / 2, -c, one], axis=-1)
+        a_half = jnp.stack(
+            [c4 / 3840, -c3 / 384, c2 / 48, -c / 8, one / 2, zero], axis=-1
+        )
+        a1 = jnp.stack([c4 / 120, -c3 / 24, c2 / 6, -c / 2, one, zero], axis=-1)
+        aa = jnp.stack([-c5 / 720, c4 / 120, -c3 / 24, c2 / 6, -c / 2, one], axis=-1)
+        b_half = jnp.stack(
+            [c4 / 46080, -c3 / 3840, c2 / 384, -c / 48, one / 8, zero], axis=-1
+        )
+        b1 = jnp.stack([c4 / 720, -c3 / 120, c2 / 24, -c / 6, one / 2, zero], axis=-1)
+        chh = jnp.stack([c4 / 168, -c3 / 30, 3 * c2 / 20, -c / 2, one, zero], axis=-1)
+        ckk = jnp.stack([5 * c4 / 168, -c3 / 7, c2 / 2, -c, zero, zero], axis=-1)
+
+        correct_shape = c.shape + (6,)
+        assert (
+            beta_half.shape
+            == a_half.shape
+            == b_half.shape
+            == beta1.shape
+            == a1.shape
+            == b1.shape
+            == aa.shape
+            == chh.shape
+            == ckk.shape
+            == correct_shape
+        )
+
+        return _ShOULDCoeffs(
+            beta_half=beta_half,
+            a_half=a_half,
+            b_half=b_half,
+            beta1=beta1,
+            a1=a1,
+            b1=b1,
+            aa=aa,
+            chh=chh,
+            ckk=ckk,
+        )
+
+    def _compute_step(
+        self,
+        h: RealScalarLike,
+        levy: AbstractSpaceTimeTimeLevyArea,
+        x0: UnderdampedLangevinX,
+        v0: UnderdampedLangevinX,
+        uld_args: ULDArgs,
+        coeffs: _ShOULDCoeffs,
+        rho: UnderdampedLangevinX,
+        prev_f: UnderdampedLangevinX,
+    ) -> tuple[UnderdampedLangevinX, UnderdampedLangevinX, UnderdampedLangevinX, None]:
+        dtypes = jtu.tree_map(jnp.result_type, x0)
+        w: UnderdampedLangevinX = jtu.tree_map(jnp.asarray, levy.W, dtypes)
+        hh: UnderdampedLangevinX = jtu.tree_map(jnp.asarray, levy.H, dtypes)
+        kk: UnderdampedLangevinX = jtu.tree_map(jnp.asarray, levy.K, dtypes)
+
+        gamma, u, f = uld_args
+
+        rho_w_k = (rho**ω * (w**ω - 12 * kk**ω)).ω
+        uh = (u**ω * h).ω
+
+        f0 = prev_f
+        v1 = (v0**ω + rho**ω * (hh**ω + 6 * kk**ω)).ω
+        x1 = (
+            x0**ω
+            + coeffs.a_half**ω * v1**ω
+            + coeffs.b_half**ω * (-(uh**ω) * f0**ω + rho_w_k**ω)
+        ).ω
+        f1 = f(x1)
+
+        chh_hh_plus_ckk_kk = (coeffs.chh**ω * hh**ω + coeffs.ckk**ω * kk**ω).ω
+
+        x_out = (
+            x0**ω
+            + coeffs.a1**ω * v0**ω
+            - uh**ω * coeffs.b1**ω * (1 / 3 * f0**ω + 2 / 3 * f1**ω)
+            + rho**ω * (coeffs.b1**ω * w**ω + chh_hh_plus_ckk_kk**ω)
+        ).ω
+        f_out = f(x_out)
```

Comment:
Same comment, in a follow-up we could consider switching this to use a scan trick.